### PR TITLE
add lognl for hierarchical models to output files

### PR DESCRIPTION
**Pull request by ahnitz**
This enables people to make plots with z-arg snr again with hierarchical based likelihoods. For the regular hierarchical it adds the lognl from each submodel (assumption of independent noise). For multisignal, it picks one (assumption of identical noise). These assumptions match the models, so I think this is correct. It also is run in a try/except in cases where a submodel for its own purposes can't produce a lognl. 

@WuShichao This should enable you to plot with SNR for your tests. Please give it a try! (requires running the inference with this patch included)

### CODE REVIEW

File: pycbc/inference/models/hierarchical.py
Comment (WuShichao) [inline pycbc/inference/models/hierarchical.py:1]: This solution is better than mine. I'm running my test by using this patch. Can you explain more about "For multisignal, it picks one (assumption of identical noise)"? If there are two overlapping signals in the same data, although the detector noise n(t) is the same, the total noise n(t) + hj(t) is different for each, right?

**Comment by ahnitz** [comment]
@WuShichao Can this now be merged?

**Comment by WuShichao** [comment]
@ahnitz Yes, let me merge it with **squash**.

**Comment by ahnitz** [comment]
@WuShichao You don't need to merger. You need to approve a review.

**Review by WuShichao — APPROVED** [review]

**Pull request merged**